src/singletons/time-manager.ts

```diff
@@ -207,13 +203,27 @@ export class TimeManager {
     // This function only applies when datetime plugin is enabled
     if (settingsManager.plugins.datetime) {
       if (this.lastTime - this.simulationTimeObj.getTime() < <Milliseconds>300) {
-        this.simulationTimeSerialized_ = this.simulationTimeObj.toJSON();
-        this.timeTextStr = this.timeTextStrEmpty_;
-        for (this.iText = 11; this.iText < 20; this.iText++) {
-          if (this.iText > 11) {
-            this.timeTextStr += this.simulationTimeSerialized_[this.iText - 1];
-          }
-        }
+        this.timeTextStr = this.simulationTimeObj.toLocaleDateString('en-US', {
+          year: '2-digit', // Fixing to 2-digits will avoid layout shifts
+          month: '2-digit',
+          day: '2-digit',
+          hour: '2-digit',
+          minute: '2-digit',
+          second: '2-digit',
+          hour12: false, // 12 hour or 24 hour time?
+          timeZoneName: 'short',
+        });
+
+        // Remove the comma separating the date and time to keep it short
+        this.timeTextStr = this.timeTextStr.replace(',', '');
+
+        // this.simulationTimeSerialized_ = this.simulationTimeObj.toJSON();
+        // this.timeTextStr = this.timeTextStrEmpty_;
+        // for (this.iText = 11; this.iText < 20; this.iText++) {
+        //   if (this.iText > 11) {
+        //     this.timeTextStr += this.simulationTimeSerialized_[this.iText - 1];
+        //   }
+        // }
```

Comment:
This section is overly complicated to allow changing the date/time. Since you aren't doing that, this can be simplified. I included the code for displaying the date/time in local time along with a time zone. You can adjust to fit whatever style you are looking for.

I did not like the comma it adds by default so I included the code to remove it.

All other changes in this file were just removing dead code.